File: C:C++/Stack.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node
{
    int data;
    Node *next;
}Node;

typedef struct Stack
{
    Node *top;
}Stack;

Stack *createStack()
{
    Stack *s;
    s = (Stack *)malloc(sizeof(Stack));
    s->top = NULL;
    return s;
}

int pop(Stack *s)
{
    if (empty(s))
        printf("Stack is Empty\n");
    else {
        Node *temp = s->top;
        int num = temp->data;
        s->top = temp->next;
        free(temp);
        return num;
    }
}

void push(Stack *s, int element)
{
    Node *temp = malloc(sizeof(Node));
    temp->data = element;
    temp->next = s->top;
    s->top = temp;
}

bool empty(Stack *s) {
    
    return (s->top == NULL);
}

File: Data Structure/Stack.java
public class Stack<Item> {

	private Node top;
    private int N;
    
    private class Node {
        Item item;
        Node next;
    }

	public Stack() {
		this.top = null;
        this.N = 0;
	}
    
    public void push(Item item) {
		Node temp = new Node();
        temp.item = item;
		temp.next = this.top;
		this.top = temp;
        N++;
	}

	public Item pop() {
        if(this.isEmpty())
            return null;
        Node temp = this.top;
        Item item = temp.val;
        this.top = temp.next;
        N--;
        return item;
	}
	
	boolean isEmpty() {
		return (this.top == null);
	}
    
    public int size() {
        return N;
    }

}
